### ISSUE TITLE
Zend_Http_Client Header with integer value throws exception

### ISSUE THREAD

**Issue by cliffordwhansen** [zendframework/zf1#566]
I just updated to 1.12.12 using composer and got the following error:
Fatal error: Uncaught exception 'Zend_Http_Exception' with message 'Invalid header value detected' in /vagrant/vendor/zendframework/zendframework1/library/Zend/Http/Client.php on line 1597

After some debugging I found the following fixes it
```
diff --git a/library/Zend/Http/Client.php b/library/Zend/Http/Client.php
index 279556a..acbbb33 100644
--- a/library/Zend/Http/Client.php
+++ b/library/Zend/Http/Client.php
@@ -1592,7 +1592,7 @@ class Zend_Http_Client
             return;
         }
 
-        if (! is_string($value) && (! is_object($value) || ! method_exists($value, '__toString'))) {
+        if (!(is_string($value)||is_integer($value)) && (! is_object($value) || ! method_exists($value, '__toString'))) {
             require_once 'Zend/Http/Exception.php';
             throw new Zend_Http_Exception('Invalid header value detected');
         }
```

My PHP version.
```
$ php -v
PHP 5.5.24-1+deb.sury.org~trusty+1 (cli) (built: May  2 2015 09:47:33) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2015 Zend Technologies
    with Zend OPcache v7.0.4-dev, Copyright (c) 1999-2015, by Zend Technologies
    with Xdebug v2.3.2, Copyright (c) 2002-2015, by Derick Rethans
```

**Comment by dirmax** [comment]
Hotfix!!!
replace string 1595
from: if (! is_string($value) && (! is_object($value) || ! method_exists($value, '__toString'))) {
to: if (! is_string($value) && **! is_numeric($value) &&** (! is_object($value) || ! method_exists($value, '__toString'))) {

P.S. Who make pull request? )

**Comment by akrabat** [comment]
PR #567 is ready to go. Hopefully we'll roll out 1.12.13 later today.

**Comment by froschdesign** [comment]
Fixed with #567
